src/opening-period/form/OpeningPeriodForm.tsx

```diff
@@ -0,0 +1,322 @@
+import React, { useState } from 'react';
+import { ArrayField, useFieldArray, useForm } from 'react-hook-form';
+import { useHistory } from 'react-router-dom';
+import {
+  DatePeriod,
+  DatePeriodOptions,
+  FormWeekdays,
+  Language,
+  ResourceState,
+  ResourceStateApiOption,
+  ResourceStateOption,
+  TimeSpan as TimeSpanApiFormat,
+  TimeSpanFormFormat,
+} from '../../common/lib/types';
+import { transformToApiFormat } from '../../common/utils/date-time/format';
+import Datepicker from '../../components/datepicker/Datepicker';
+import { PrimaryButton, SecondaryButton } from '../../components/button/Button';
+import { ErrorToast, SuccessToast } from '../../components/notification/Toast';
+import OpeningPeriodDescription from '../description/OpeningPeriodDescription';
+import TimeSpan from '../time-span/TimeSpan';
+import './OpeningPeriodForm.scss';
+
+type OpeningPeriodDataBase = {
+  openingPeriodTitle: string;
+  openingPeriodOptionalDescription: string;
+  openingPeriodBeginDate: string | undefined;
+  openingPeriodEndDate: string | undefined;
+};
+
+interface OpeningPeriodFormData extends OpeningPeriodDataBase {
+  timeSpans: Array<TimeSpanFormFormat> | {}[];
+}
+
+export interface OpeningPeriodApiData extends OpeningPeriodDataBase {
+  timeSpans: Array<TimeSpanApiFormat>;
+}
+
+type SubmitStatus = 'init' | 'succeeded' | 'error';
+
+type NotificationTexts = {
+  label: string;
+  description: string;
+};
+
+function formatTimeSpansToApiFormat(
+  timeSpans: TimeSpanFormFormat[]
+): TimeSpanApiFormat[] {
+  return timeSpans.map((timeSpan) => {
+    return {
+      description: {
+        fi: timeSpan?.description,
+        sv: null,
+        en: null,
+      },
+      end_time: `${timeSpan.endTime}:00`,
+      start_time: `${timeSpan.startTime}:00`,
+      resource_state: timeSpan.resourceState,
+      weekdays: timeSpan.weekdays.reduce(
+        (acc: Array<number>, currentValue: boolean, currentIndex: number) => {
+          if (currentValue) {
+            acc.push(currentIndex + 1);
+            return acc;
+          }
+          return acc;
+        },
+        []
+      ),
+    };
+  });
+}
+
+const dropMilliseconds = (time: string): string => time.slice(0, -3);
+const getLowestWeekdayNumber = (timeSpan: TimeSpanApiFormat): number =>
+  timeSpan && timeSpan.weekdays ? timeSpan.weekdays.sort()[0] : 0;
+
+function formatApiTimeSpansToFormFormat(
```

Comment:
I wonder if it would be a good idea to add unit tests for these formatFunctions `formatApiTimeSpansToFormFormat`and `formatTimeSpansToApiFormat`. They seem like functions that would be easy and handy to unit test.

Reply:
I'll add a tests. I may have to extract these

Reply:
Extracted and added tests